control-base/src/main/java/fi/nls/oskari/control/layer/SaveLayerHandler.java

```diff
@@ -270,13 +323,21 @@ private boolean handleRequestToMapLayer(final ActionParameters params, OskariLay
             }
         }
 
-        // FIXME: Update JSON key when frontend implementation was updated
-        MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(params.getHttpParam("inspireTheme", -1));
-        ml.addGroup(maplayerGroup);
-
-        ml.setVersion(params.getHttpParam("version", ""));
+        String groupId = params.getHttpParam(PARAM_MAPLAYER_GROUPS, "-1");
+        ml.emptyMaplayerGroups();
```

Comment:
Dataprovider is added as a group before this so you need to empty the array here? You should refactor this: 
 +        final DataProvider group = dataProviderService.find(params.getHttpParam(PARAM_GROUP_ID, -1));
          ml.addGroup(group);

Is dataprovider saved at all?

Reply:
This is confusing because of the naming. `DataProvider group` should be renamed to `DataProvider dataProvider`. In addition, `private Set<DataProvider> groups` inside OskariLayer should be renamed to `private Set<DataProvider> dataProviders` and `OskariLayer#addGroup(final DataProvider group)` should be renamed `OskariLayer#addDataProvider(final DataProvider dataProvider)`

Reply:
Please do this renaming ^